### PR TITLE
fix: could not get parameter from `design:paramtypes` metadata key

### PR DESCRIPTION
**Pull request by chhoangcmc**
Hi @alphamikle ,
Thank you very much for this helpful package!
Currently I have encountered an issue when developing my application.

### TLDR;
The problem here is that `nest-transact` does not have logic to handle dependency of configuration objects from `ConfigModule `([Documentation | NestJS - A progressive Node.js framework](https://docs.nestjs.com/techniques/configuration) ).

### What happens?
When we wrap a NestJS service with `nest-transact` with the usage of `withTransact` function, the dependency of configuration objects (`ConfigType`) which we get from `Reflect.getMetadata('design:paramtypes', constructor)` would be `undefined`, and `nest-transact` does not have logic to handle this kind of dependency, causing the application to be crashed.

The dependency of configuration object does exist, but with other metadata key `'self:paramtypes'`.
image

image

### Troubleshooting
I have created this PR for the fix. Please take a look and I am looking forward to your approval. Many thanks 🙏

### CODE REVIEW

**Comment by chhoangcmc** [comment]
Sorry that I could not assign the reviewer :( Please take a look, @alphamikle

**Comment by chhoangcmc** [comment]
Hi @alphamikle , thank you for the approval of the merge!
Could you please publish the new version of `nest-transact`?

**Comment by alphamikle** [comment]
Hi @chhoangcmc! Version 9.1.2 is on the market. Thank you for your PR!
